imap_processing/ultra/decom_ultra.py

```diff
@@ -0,0 +1,160 @@
+# Standard
+import logging
+
+import numpy as np
+
+# Installed
+import xarray as xr
+
+# Local
+from imap_processing import decom
+
+logging.basicConfig(level=logging.INFO)
+#TODO: This is a work in progress. Hard-coded values will be removed.
+def read_n_bits(binary: str, n: int, current_position: int):
+    """
+    Extract a specified number of bits from a binary string,
+    starting from a given position.
+
+    Parameters
+    ----------
+    binary : str
+        The binary string from which bits will be read.
+    n : int
+        Number of bits to read from the binary string.
+    current_position : int
+        The starting position in the binary string from which bits will be read.
+
+    Returns
+    -------
+    value : int
+        The integer representation of the read bits or None if the end of the
+        string is reached before reading 'n' bits.
+    current_position + n
+        - The updated position in the binary string after reading the bits.
+    """
+
+    # Ensure we don't read past the end
```

Comment:
raise IndexError to indicate that we read past the end? Otherwise this is sort of hidden and we may not know we implemented something wrong.